bin/oneoff/backfill_data/backfill_section_course_id.rb

```diff
@@ -0,0 +1,59 @@
+#!/usr/bin/env ruby
+
+require_relative '../../../dashboard/config/environment'
+
+# This script backfills the section table to add course_ids to old sections that have a script but no course.
+
+CDO.log = Logger.new($stdout)
+ActiveRecord::Base.record_timestamps = false
+
+options = {actually_update: false, start_with: 1}
+OptionParser.new do |opts|
+  opts.banner = "Usage: #{File.basename(__FILE__)} [options]"
+  opts.on('-u', '--actually-update', 'Actually perform the update.') do
+    options[:actually_update] = true
+  end
+  opts.on('-sSTART', '--start-with=START', Integer, 'Start with section ID START.') do |s|
+    options[:start_with] = s
+  end
+  opts.on('-h', '--help', 'Add -u to perform the update.') do
+    puts opts
+    exit
+  end
+end.parse!
+CDO.log.info "Called with options: #{options}"
+
+BATCH_SIZE = 100
+INFO_INTERVAL = 1_000
+
+sections_processed = 0
+sections_not_processed = 0
+
+Section.with_deleted.where(course_id: nil).where.not(script_id: nil).find_each(batch_size: BATCH_SIZE, start: options[:start_with]) do |section|
+  # Be kind to the database by limiting to 1000 sections processed per second
```

Comment:
remove outdated numbers from comment? it looks like 100,000 sections / sec to me.